components/core/src/clp_s/RangeIndexWriter.cpp

```diff
@@ -0,0 +1,88 @@
+#include "RangeIndexWriter.hpp"
+
+#include <cstddef>
+#include <cstdint>
+#include <exception>
+#include <system_error>
+#include <utility>
+#include <vector>
+
+#include <nlohmann/json.hpp>
+#include <nlohmann/json_fwd.hpp>
+#include <outcome/outcome.hpp>
+
+#include "ErrorCode.hpp"
+#include "SingleFileArchiveDefs.hpp"
+#include "ZstdCompressor.hpp"
+
+namespace clp_s {
+auto RangeIndexWriter::open_range(size_t start_index)
+        -> OUTCOME_V2_NAMESPACE::std_result<handle_t> {
+    for (auto const& range : m_ranges) {
+        if (start_index < range.start_index || false == range.end_index.has_value()
+            || start_index < range.end_index.value())
+        {
+            return std::errc::invalid_argument;
+        }
+    }
```

Comment:
Do we want to print an error message in this case (and probably better error code)?